browser/src/control/jsdialog/Widget.TreeView.ts

```diff
@@ -521,6 +546,92 @@ class TreeViewControl {
 		}
 	}
 
+	/**
+	 * Creates a partially highlighted cell for the tree view.
+	 * Highlighted part of text corresponds to where core has marked a hit
+	 *
+	 * e.g. searching for 'line', core sends the following:
+	 * 		Collabora On[line],
+	 * 		https://www.collaboraon[line].com,
+	 * 		des graphiques spark[line],
+	 *      etc.
+	 */
+	createHighlightedCell(
+		parent: HTMLElement,
+		entry: TreeEntryJSON,
+		index: any,
+		builder: JSBuilder,
+		searchTerm: string,
+	) {
+		const sourceText = entry.text;
+
+		const searchPattern = `[${searchTerm}]`;
+		const mainSpan = L.DomUtil.create(
+			'span',
+			builder.options.cssClass + ` ui-treeview-cell-text-content`,
+			parent,
+		);
+
+		const fragments = this.caseInsensitiveSplit(sourceText, searchPattern);
+		if (fragments.length === 1) {
+			/// not found
+			mainSpan.appendChild(document.createTextNode(sourceText));
+		} else {
+			// found, can be many times
+			for (let i = 0; i < fragments.length - 1; i += 2) {
+				mainSpan.appendChild(document.createTextNode(fragments[i])); // pre
+				const highlightSpan = L.DomUtil.create(
+					'span',
+					builder.options.cssClass + ' highlighted',
+					mainSpan,
+				);
+				highlightSpan.innerText = fragments[i + 1].substring(
+					1,
+					fragments[i + 1].length - 1,
+				);
+				mainSpan.appendChild(highlightSpan);
+			}
+			mainSpan.appendChild(
+				document.createTextNode(fragments[fragments.length - 1]),
+			); // post
+		}
+
+		return mainSpan;
+	}
+
+	isPageDivider(
+		entry: TreeEntryJSON,
+		pageEntryPrefix: string,
+		pageEntrySuffix: string,
+	): boolean {
+		// Matches page divider prefix and suffix: -$#~ PAGE ~#$- as set in core: QuickFindPanel::FillSearchFindsList() (QuickFindPanel.cxx)
+		return (
+			entry.text &&
+			entry.text.startsWith(pageEntryPrefix) &&
+			entry.text.endsWith(pageEntrySuffix)
+		);
+	}
+
+	getPageEntryText(
+		text: string,
+		pageEntryPrefix: string,
+		pageEntrySuffix: string,
+	): string {
+		return text.substring(
+			pageEntryPrefix.length,
+			text.length - pageEntrySuffix.length,
+		);
+	}
+
+	caseInsensitiveSplit(text: string, delimeter: string) {
+		// escape regex special chars
+		const escapedPattern = delimeter.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
```

Comment:
Note for myself: we need to move such thing to some global util for future.
We have only HTML sanitizer so far.